integrations/pgvector/src/haystack_integrations/document_stores/pgvector/filters.py

```diff
@@ -0,0 +1,228 @@
+
+    return where_clause, params
+
+
+def _parse_logical_condition(condition: Dict[str, Any]) -> tuple[str, List[Any]]:
+    if "operator" not in condition:
+        msg = f"'operator' key missing in {condition}"
+        raise FilterError(msg)
+    if "conditions" not in condition:
+        msg = f"'conditions' key missing in {condition}"
+        raise FilterError(msg)
+
+    conditions = [_parse_comparison_condition(c) for c in condition["conditions"]]
+    query_parts, values = [], []
+    for c in conditions:
+        query_parts.append(c[0])
+        values.append(c[1])
+    if isinstance(values[0], list):
+        values = list(chain.from_iterable(values))
+
+    operator = condition["operator"]
+    if operator == "AND":
+        sql_query = "(" + " AND ".join(query_parts) + ")"
+    elif operator == "OR":
+        sql_query = "(" + " OR ".join(query_parts) + ")"
+    else:
+        msg = f"Unknown logical operator '{operator}'"
+        raise FilterError(msg)
+
+    return sql_query, values
+
+
+def _parse_comparison_condition(condition: Dict[str, Any]) -> tuple[str, List[Any]]:
+    if "field" not in condition:
+        # 'field' key is only found in comparison dictionaries.
+        # We assume this is a logic dictionary since it's not present.
+        return _parse_logical_condition(condition)
+
+    field: str = condition["field"]
+    if "operator" not in condition:
+        msg = f"'operator' key missing in {condition}"
+        raise FilterError(msg)
+    if "value" not in condition:
+        msg = f"'value' key missing in {condition}"
+        raise FilterError(msg)
+
+    operator: str = condition["operator"]
+    value: Any = condition["value"]
+    if isinstance(value, DataFrame):
+        # DataFrames are stored as JSONB and we query them as such
+        value = value.to_json()
+        field = f"({field})::jsonb"
+        value = Jsonb(value)
+
+    if field.startswith("meta."):
+        field = _treat_meta_field(field, value)
+
+    field, value = COMPARISON_OPERATORS[operator](field, value)
+    return field, [value]
+
+
+def _treat_meta_field(field: str, value: Any) -> str:
+    """
+    Internal method that modifies the field str
+    to make the meta JSONB field queryable.
+    """
+
+    # use the ->> operator to access keys in the meta JSONB field
+    field_name = field.split(".", 1)[-1]
+    field = f"meta->>'{field_name}'"
+
+    # meta fields are stored as strings in the JSONB field,
+    # so we need to cast them to the correct type
+    type_value = PYTHON_TYPES_TO_PG_TYPES.get(type(value))
+    if isinstance(value, list) and len(value) > 0:
+        type_value = PYTHON_TYPES_TO_PG_TYPES.get(type(value[0]))
+
+    if type_value:
+        field = f"({field})::{type_value}"
+
+    return field
+
+
+def _equal(field: str, value: Any) -> tuple[str, Any]:
+    if value is None:
+        # no_value is a placeholder that will be removed in _convert_filters_to_where_clause_and_params
```

Comment:
If that's the case, let's declare it as a constant `NO_VALUE="no_value"` at the module level so it's easier to change if needed